### ISSUE TITLE
Endless Method Keyword/Operator Precedence Check

### ISSUE THREAD

**Issue by phene** [rubocop/rubocop#12988]
## Is your feature request related to a problem?

Consider the following code

```ruby
def foo = true and false
def bar = true && false

foo # => true
bar # => false
```

Because the `=` operator has higher precedence than `and` or `or`, they will not be included as part the method.

## Describe the solution you'd like

A rubocop check to detect operators/keywords with lower precedence than `=` on the same line as an endless expression would be incredibly helpful in avoiding this sort of footgun.

**Comment by vlad-pisanov** [comment]
Such a cop is badly needed. Our team has run into `if`/`unless` gotchas several times:

this doesn't work:
```ruby
def foo = 42 if cond
# NoMethodError: undefined method `cond' for main
```

but this works fine: 🙄 
```ruby
def foo(cond) = 42 if cond
:foo
```

**Comment by vlad-pisanov** [comment]
@dvandersluis I haven't encountered this pattern before. Maybe it could be conditionally permitted for endless methods with a cop option like `AllowInlineModifiers` or something? 🤔

**Comment by dvandersluis** [comment]
Perhaps I'm misremembering it's prevalance. A github code search reveals some uses but not as prevalent as I thought. There's also `unless respond_to?` in any case. 

I think I'll just leave it for now and if issues come up we can address.

**Comment by phene** [comment]
FWIW, `def foo = 42 unless defined?(:foo)` seems like a major anti-pattern.
